ukpn/scripts/download_data.py

```diff
@@ -0,0 +1,198 @@
+                ) is True:
+                    pv_site_dict_index.append(i)
+
+    # CHecking if there are any sites matching the coordinates
+    if len(pv_site_dict_index) == 0:
+        logger.info(f"There are no PV sites matching with eastinngs: {eastings}")
+        logger.info(f"There are no PV sites matching with northings: {northings}")
+        return None
+    else:
+        # Getting the required data from Eastings and Northings
+        data_json = data_records[pv_site_dict_index[0]]
+
+        return data_json
+
+
+def get_metadata_from_ukpn_xlsx(
+    link_of_ecr_excel: str,
+    local_path: Path[Union, str],
+    eastings: Optional[str] = None,
+    northings: Optional[str] = None,
+) -> pd.DataFrame:
+    """Download and load the ECR file from the link provided below
+
+    For direct download, opne this link-
+    https://media.umbraco.io/uk-power-networks/0dqjxaho/embedded-capacity-register.xlsx
+
+    Args:
+        link_of_ecr_excel: Link shown above
+        local_path: The folder where the file needs to get downloaded
+        eastings: eastings value of the pv solar farm
+        northings: Northings value of the pv solar farm
+    """
+    # Download and store the excel sheet in a location
+    resp = requests.get(link_of_ecr_excel)
+    local_path = os.path.join(local_path, "ecr.xlsx")
+    with open(local_path, "wb") as output:
+        output.write(resp.content)
+
+    # Read the excel sheet
+    wb = load_workbook(local_path, read_only=True, keep_links=False)
+
+    # The sheet need and its name according to UKPN is "Register Part 1"
+    file_name = "Register Part 1"
+    for text in wb.sheetnames:
+        if file_name in text:
+            df = pd.read_excel(local_path, sheet_name=text, skiprows=1)
+
+    # Filtering the data based on the eastings and northings provided
+    for text in df.columns:
+        if "Eastings" in text:
+            df = df[df[text] == np.float64(eastings)].reset_index()
+
+    return df
+
+
+def metadata_df_to_netcdf(
+    path_to_ukpn_timeseries: str, input_dataframe: Optional[pd.DataFrame] = None
+) -> xr.Dataset:
+
+    # Loading the UKPN time series data into a dataframe
+    original_df = load_csv_to_pandas(path_to_file=path_to_ukpn_timeseries)
+
+    # Interpolating time series
+    interpolated_df = interpolation_pandas(original_df=original_df)
+
+    # Getting all the time series dates
+    interpolated_timeseries = interpolated_df.index.values
+    interpolated_data_values = interpolated_df[interpolated_df.columns[0]].values
+
+    # Creating an Xarray data array
+    final_xarray = xr.DataArray(
+        data=interpolated_data_values,
+        dims="time_utc",
+        coords={"time_utc": interpolated_timeseries},
+        attrs={"Description": " This Data array consists of time-series data from UKPN"},
+    )
+
+    # Getiing the column names
+    # TODO
+    required_column_names = ["Eastings", "Northings", "Maximum Export Capacity"]
```

Comment:
Potentially change the columns in the DataArray to more match the other NetCDF training files 
```suggestion
    required_column_names = ["x_osgb", "y_osgb", "capacity_megawatt_power"]
```

Reply:
sure, I'll use the attribute names mentioned in the other `NetCDF` files!